src/coreclr/gc/gc.cpp

```diff
@@ -21507,10 +21507,13 @@ int gc_heap::generation_to_condemn (int n_initial,
     }
 
 #ifdef USE_REGIONS
-    if (!try_get_new_free_region())
+    if (!check_only_p)
```

Comment:
Using `&&` to avoid changing many lines?

Reply:
That annoyed me a bit too, but I followed the pattern at line 21544 instead.  I assumed that it was to make it clear by separating the "mode" of the function and the check itself, which made sense to me.